test/e2e/clusterctl_upgrade.go

```diff
@@ -217,7 +217,7 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 		Expect(os.MkdirAll(input.ArtifactFolder, 0750)).To(Succeed(), "Invalid argument. input.ArtifactFolder can't be created for %s spec", specName)
 
 		// Setup a Namespace where to host objects for this spec and create a watcher for the namespace events.
-		managementClusterNamespace, managementClusterCancelWatches = setupSpecNamespace(ctx, specName, input.BootstrapClusterProxy, input.ArtifactFolder, input.PostNamespaceCreated)
+		managementClusterNamespace, managementClusterCancelWatches = setupSpecNamespace(ctx, specName, input.BootstrapClusterProxy, input.ArtifactFolder, nil)
```

Comment:
Don't we want to call it in both places?

Reply:
I would say not for now, see https://github.com/kubernetes-sigs/cluster-api/pull/10067#discussion_r1469963592

Reply:
Didn't see your comment. I'll take a look

Reply:
after chatting about this with @sbueringer, in the clusterctl-upgrade and in the self-hosted tests we are calling the func both for the newly created management cluster and the workload cluster